ecommerce/baskets/views.py

```diff
@@ -15,12 +21,61 @@ class BasketItemViewSet(DetailedViewSetMixin, viewsets.ModelViewSet):
         "detailed": BasketItemDetailedSerializer,
     }
 
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        user_id = self.request.user.id
+        return queryset.filter(basket__customer__id=user_id)
+
 
 class BasketViewSet(DetailedViewSetMixin, viewsets.ModelViewSet):
-    queryset = Basket.objects.all()
-    serializer_class = BasketSerializer
-    filterset_class = BasketFilter
+    http_method_names = ["get", "delete"]
```

Comment:
basket'i kullaniciya sildirmek guzel bir fikir olmayabilir.

Reply:
Basketi silme amaçlı değil. BasketItemı silme amaçlı yapıldı. Kullanıcı sepetindeki ürünleri sepetinden kaldırabilir.

Reply:
BasketViewSet bu sinifin altinda is yaptigimizi dusunerek Basket uzerinde calistigimizi dusunuyorum. Typo mu oldu sorusunu da o yuzden sormustum.

Reply:
Hocam yine Basket üzerinde çalışıyoruz ama işlemleri BasketItem üzerinden yapıyoruz. Sırasıyla Basket yoksa Basket oluştur, Basket varsa BasketItemları eklenebilsin. Silinmek istendiğinde BasketItemlar silinebilsin. Yani burdaki delete bir nevi BasaketItem update işlevi.

Reply:
genel yaklasimda hangi model uzerinde islem yapiyorsak viewset ismimiz o olursa guzel olur. 

Katiliyorum, BasketItem i silme yaklasimi kotu bir fikir degil.